app/vmagent/remotewrite/client.go

```diff
@@ -481,22 +480,37 @@ again:
 var remoteWriteRejectedLogger = logger.WithThrottler("remoteWriteRejected", 5*time.Second)
 
 type rateLimiter struct {
-	perSecondLimit int64
+	interval         time.Duration
+	perIntervalLimit int64
+	stopCh           <-chan struct{}
 
 	// mu protects budget and deadline from concurrent access.
 	mu sync.Mutex
 
-	// The current budget. It is increased by perSecondLimit every second.
+	// The current budget. It is increased by perIntervalLimit every interval.
 	budget int64
 
-	// The next deadline for increasing the budget by perSecondLimit
+	// The next deadline for increasing the budget by perIntervalLimit
 	deadline time.Time
 
 	limitReached *metrics.Counter
 }
 
-func (rl *rateLimiter) register(dataLen int, stopCh <-chan struct{}) {
-	limit := rl.perSecondLimit
+func newRateLimiter(interval time.Duration, perIntervalLimit int64, limitReached *metrics.Counter, stopCh <-chan struct{}) *rateLimiter {
```

Comment:
do we need interval if it's always time.second?

Reply:
I was going to bring it to separate place and i made it more generic, but changed my mind. Do you think, that should i revert interval adding?

Reply:
I think it's fine to have a change at this pr, if it'll be used lately by other caller.